common-gradle-enterprise-gradle-configuration/settings.gradle.kts

```diff
@@ -0,0 +1,36 @@
+plugins {
+    id("com.gradle.enterprise") version("3.11.1")
+    id("com.gradle.common-custom-user-data-gradle-plugin") version("1.8")
+}
+
+val isCI = !System.getenv("CI").isNullOrEmpty() // adjust to your CI provider
+
+gradleEnterprise {
+    server = "https://enterprise-samples.gradle.com" // adjust to your GE server
+    allowUntrustedServer = false // ensure a trusted certificate is configured
+
+    buildScan {
+        capture { isTaskInputFiles = true }
+        isUploadInBackground = !isCI
+        publishAlways()
+    }
+}
+
+buildCache {
+    local {
+        isEnabled = true
+    }
+
+    remote<HttpBuildCache> {
```

Comment:
shouldn't this use the new cache integration?

Reply:
I would say so :) @runningcode ?

Reply:
I agree to do this but we should do it in a separate PR.